packages/date/src/DateRange.js

```diff
@@ -158,8 +158,8 @@ const DateRange = ({
   };
 
   const onFocusChange = async input => {
-    if (!input) await setFieldTouched(name, true);
-    if (autoSync) await syncDates();
+    if (!autoSync && !input) await setFieldTouched(name, true);
+    if (autoSync && focusedInput === input) await syncDates();
     setFocusedInput(input);
```

Comment:
One last request and then I think we're good. It looks like `onDatesChange` is what was causing `focusedInput` to equal `input` and then cause a loop. If we change this to 
```
if (focusedInput !== input) {
      setFocusedInput(input); 
}
```
we can change the old line back to `if(autoSync) await syncDates();` and functionality should stay the same.

Reply:
That does look a lot clearer - thanks for all the help!